test/rapi/test_rc_api_info.c

```diff
@@ -425,6 +425,77 @@ static void test_process_fetch_game_titles_response() {
   rc_api_destroy_fetch_game_titles_response(&fetch_game_titles_response);
 }
 
+static void test_init_fetch_hash_library_request() {
+  rc_api_fetch_hash_library_request_t fetch_hash_library_request;
+  rc_api_request_t request;
+
+  memset(&fetch_hash_library_request, 0, sizeof(fetch_hash_library_request));
+  fetch_hash_library_request.console_id = 1;
+
+  ASSERT_NUM_EQUALS(rc_api_init_fetch_hash_library_request(&request, &fetch_hash_library_request), RC_OK);
+  ASSERT_STR_EQUALS(request.url, DOREQUEST_URL);
+  ASSERT_STR_EQUALS(request.post_data, "r=hashlibrary&c=1");
+  ASSERT_STR_EQUALS(request.content_type, RC_CONTENT_TYPE_URLENCODED);
+
+  rc_api_destroy_request(&request);
+}
+
+static void test_process_fetch_hash_library_response() {
+  rc_api_fetch_hash_library_response_t fetch_hash_library_response;
+  rc_api_server_response_t fetch_hash_library_server_response;
+  rc_api_hash_library_entry_t* entry;
+  const char* server_response = "{\"Success\":true,\"MD5List\":{"
+    "\"aabbccddeeff00112233445566778899\":1,"
+    "\"99aabbccddeeff001122334455667788\":1,"
+    "\"8899aabbccddeeff0011223344556677\":2"
+    "}}";
+
+  memset(&fetch_hash_library_server_response, 0, sizeof(fetch_hash_library_server_response));
+  fetch_hash_library_server_response.body = server_response;
+  fetch_hash_library_server_response.body_length = strlen(server_response);
+  fetch_hash_library_server_response.http_status_code = 200;
+
+  memset(&fetch_hash_library_response, 0, sizeof(fetch_hash_library_response));
+
+  ASSERT_NUM_EQUALS(rc_api_process_fetch_hash_library_server_response(&fetch_hash_library_response, &fetch_hash_library_server_response), RC_OK);
+  ASSERT_NUM_EQUALS(fetch_hash_library_response.response.succeeded, 1);
+  ASSERT_PTR_NULL(fetch_hash_library_response.response.error_message);
+  ASSERT_NUM_EQUALS(fetch_hash_library_response.num_entries, 3);
+
+  entry = &fetch_hash_library_response.entries[0];
+  ASSERT_NUM_EQUALS(entry->game_id, 1);
+  ASSERT_STR_EQUALS(entry->hash, "aabbccddeeff00112233445566778899");
+  entry = &fetch_hash_library_response.entries[1];
+  ASSERT_NUM_EQUALS(entry->game_id, 1);
+  ASSERT_STR_EQUALS(entry->hash, "99aabbccddeeff001122334455667788");
+  entry = &fetch_hash_library_response.entries[2];
+  ASSERT_NUM_EQUALS(entry->game_id, 2);
+  ASSERT_STR_EQUALS(entry->hash, "8899aabbccddeeff0011223344556677");
+
+  rc_api_destroy_fetch_hash_library_response(&fetch_hash_library_response);
+}
+
+static void test_process_fetch_hash_library_empty_response() {
+  rc_api_fetch_hash_library_response_t fetch_hash_library_response;
+  rc_api_server_response_t fetch_hash_library_server_response;
+  const char* server_response = "{\"Success\":true,\"MD5List\":[]}"; /* RAWeb returns a list instead of a map for invalid console */
```

Comment:
I plan to correct that, but it's good to ensure the current behavior is functional.